### ISSUE TITLE
Explore new patterns for Learn more buttons and expandable section buttons for all applications

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#1817]
## Description

The GI Bill Comparison Tool® has an open issue where `Learn more` buttons are [interfering with screen reader behavior](https://github.com/department-of-veterans-affairs/va.gov-team/issues/1251) on input labels. This lead to a bit more exploration into buttons in labels and legends to understand how they affected screen readers. 

This ticket will collect 508 findings and be a point of coordination with design to clarify and create guidance for sitewide patterns.

## Point of Contact

**VFS Point of Contact:** _Kevin (design), Trevor (508)_

## Acceptance Criteria

As a screen reader user:
* I want to be hear `<legend>` text for checkboxes and radio buttons when appropriate
* I do not want the button behavior to interfere with click/keypress behavior on input labels
* I want to be able to click/keypress the `<button>` to expand content areas or launch modal windows

## Environment

* Windows 10
* IE11, Chrome
* JAWS
* https://staging.va.gov/gi-bill-comparison-tool (button in label)
* https://staging.va.gov/gi-bill-comparison-tool/profile/11806124 (button in legend)

## Possible Fixes (optional)

Codepen link forthcoming

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2019-09-16 at 4 42 51 PM](https://user-images.githubusercontent.com/934879/64996175-dae12300-d8a2-11e9-99a9-4a7a4001fff4.png)

---
![Screen Shot 2019-09-16 at 4 42 20 PM](https://user-images.githubusercontent.com/934879/64996187-e2a0c780-d8a2-11e9-9737-93bb2c36ab53.png)

---
![Screen Shot 2019-09-13 at 1 26 32 PM](https://user-images.githubusercontent.com/934879/64996196-e896a880-d8a2-11e9-938b-9f0c7ab8afca.png)

**Comment by 1Copenut** [comment]
@KevinMHoffmanUSDS & @jenstrickland I added what I've got to Codepen ( https://codepen.io/tpierce_402/pen/PoYVKKa ) and added it to the ticket description. It's been a few days since I tested with assistive devices, so that would be a good idea to get a benchmark for usability of these patterns.

I can say the button in the input label was especially tricky. Basically the button has to be after the input, and be treated as a block-level element to make screen readers read it separate from the input label.

I'll follow up when I'm back and see how I can help.

**Comment by 1Copenut** [comment]
@jenstrickland && @KevinMHoffmanUSDS I'd like to schedule some time to discuss this item this week. I think it'll be a design pattern update, so feel an hour meeting would be in order to discuss findings, and possible UI changes.

**Comment by KevinMHoffmanUSDS** [comment]
@1Copenut Sounds good! I've got time Wednesday Oct. 9 before noon ET or between 1 and 3pm ET.

**Comment by 1Copenut** [comment]
@amycesal Here's the issue I was talking about earlier. We're testing this solution with screen readers now, and the preliminary results look good.

**Comment by 1Copenut** [comment]
@andreahewitt-odd I'm going to move this into `Review` for the VSP board. I've added about as much value as I can at this point, and waiting to see if there's a need for further coded versions like a simpler tooltip.

**Comment by 1Copenut** [comment]
Add a tooltip-like short explainer and guidance for when to use it over the more interactive Learn more elements.

**Comment by peggygannon** [comment]
@1Copenut Hi, Trevor, I wanted to clarify what's needed from me on this issue. Are there specific "Learn more" links that need tooltips written? If so, it would be helpful if we track these tooltips in a separate Content issue.

**Comment by jenstrickland** [comment]
@peggygannon Hi Peggy, we hoped for an evaluation of the "Learn more" label/words. It seems 'antiseptic' and we thought there might be another approach to resonate with the Veterans language. For screenreaders we're adding more context in the aria label. On some pages, there are so many "Learn more" links it introduces cognitive confusion. While looking for solutions to address the screenreader accessibility issues we noticed how https://calc.gsa.gov/ did it, thought it felt more human, although it would be too much for all the instances on these sample pages.

https://staging.va.gov/gi-bill-comparison-tool 
![GIBCT-Learnmore](https://user-images.githubusercontent.com/57469/67433774-be798a00-f5b6-11e9-91a0-8d2e7ae9989d.png)

**Comment by peggygannon** [comment]
@jenstrickland OK, thank you for the context. I'll take a look and suggest more descriptive links.

**Comment by jenstrickland** [comment]
@peggygannon - Thank you! 

@1Copenut Do you think testing the patterns is the next step? We discussed documenting patterns for:

- longer form additional content (the show/hide example shown in the Codepen)

- smaller amount of info patterns (either including in the label, showing on the page, or presenting in a tooltip)

**Comment by 1Copenut** [comment]
@jenstrickland Yes, testing should be our next stop. @CrystabelReiter and @emilywaggoner are working on a test plan for next VSP sprint (starts tomorrow) and I need to update the public Codepen to include a tooltip for short snippet help. I'm going to try and get that taken care of today or tomorrow.

Let's circle up and start capturing our thoughts on these different mechanisms, and when to use each. The preliminary plan is to test the Codepen, then we'll find a use case for them in VA.gov codebase, then Formation design system.

**Comment by jenstrickland** [comment]
@1Copenut For the tooltip, the implementation at https://calc.gsa.gov/ (see What is this? in the form) may be something to look at, available on hover and focus.

**Comment by 1Copenut** [comment]
@jenstrickland Codepen has been updated with tooltips: https://codepen.io/tpierce_402/pen/PoYVKKa

I opted for [Tippy.js](https://atomiks.github.io/tippyjs/) because it didn't require jQuery and had easy CDN and Node module integrations. Will make porting this to React components easier. These need screen reader testing, but the initial code is added and ready to roll.

**Comment by meganhkelley** [comment]
@1Copenut @jenstrickland do y'all need any more support from @peggygannon on this epic or ticket #4331? If not I will close #4331 and remove our team label from this epic. Let me know, thanks!

**Comment by jenstrickland** [comment]
@meganhkelley Not at this time. @peggygannon graciously provided feedback on 10/29/19. We've been in the weeds and unable to follow up further. 

I think this is currently in VSP Design's hands to design and update documentation in the design system.
